protoc-gen-eventuals/protoc-gen-eventuals.py

```diff
@@ -41,31 +41,51 @@ def process_file(self, proto_file: FileDescriptorProto):
 
         file_digest = self.analyze_file(proto_file)
 
-        proto_file_name = file_digest.pop('proto_file_name')
+        proto_file_path = file_digest.pop('proto_file_name')
+
+        index = len(proto_file_path)
+        for ch in reversed(proto_file_path):
+            if (ch == '/'):
+                break
+            index -= 1
+
+        proto_file_name = proto_file_path[index:]
 
         header_file_name = proto_file_name.replace('.proto', '.eventuals.h')
         source_file_name = proto_file_name.replace('.proto', '.eventuals.cc')
 
         eventuals_data = {
             'eventuals_header': header_file_name,
-            'grpc_pb_header': proto_file_name.replace('.proto', '.grpc.pb.h'),
+            'grpc_pb_header': '../' + proto_file_name.replace('.proto', '.grpc.pb.h'),
+            'service_number': 0,
+            'method_number': 0,
         }
 
         template_data = dict(**eventuals_data, **file_digest)
 
-        outputs = [
+        general_outputs = [
             (header_file_name, 'eventuals.h.j2'),
             (source_file_name, 'eventuals.cc.j2'),
         ]
 
-        for file_name, template_name in outputs:
-
+        # Generate only base files, without rpc methods implementation
+        for file_name, template_name in general_outputs:
             template = load_template(template_name)
             content = template.render(**template_data)
+            self.response.file.add(name=file_name, content=content)
+
+        generate_methods_template_name = 'grpc-method-eventuals.cc.j2'
+
+        for service_index, service in enumerate(template_data['services'], start = 0):
+            for method_index, method in enumerate(service['methods'], start = 0):
+                template = load_template(generate_methods_template_name)
+                template_data['service_number'] = service_index
+                template_data['method_number'] = method_index
+
+                content = template.render(**template_data)
```

Comment:
... and then change this line to:
```python
content = template.render(**template_data, **service_data)
```

Making it (hopefully very) clear that we are rendering the template based on a combination of generic data (from `template_data` and some service specific data from `service_data`.